FinniversKit/Sources/Recycling/GridViews/Markets/CompactView/CompactMarketsView.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+public class CompactMarketsView: UIView, MarketsView {
+
+    // MARK: - Private properties
+
+    private weak var delegate: MarketsViewDelegate?
+    private weak var dataSource: MarketsViewDataSource?
+
+    private lazy var collectionViewLayout: UICollectionViewLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        layout.minimumLineSpacing = .spacingS
+        layout.minimumInteritemSpacing = .spacingS
+        layout.estimatedItemSize = UICollectionViewFlowLayout.automaticSize
+        return layout
+    }()
+
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: collectionViewLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.backgroundColor = .clear
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.alwaysBounceHorizontal = true
+        collectionView.register(CompactMarketsViewCell.self)
+        collectionView.contentInset = UIEdgeInsets(vertical: 0, horizontal: .spacingM)
+        return collectionView
+    }()
+
+    // MARK: - Init
+
+    public init(
+        frame: CGRect = .zero,
+        delegate: MarketsViewDelegate,
+        dataSource: MarketsViewDataSource
+    ) {
+        super.init(frame: frame)
+
+        self.delegate = delegate
+        self.dataSource = dataSource
+
+        setup()
+    }
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+
+        addSubview(collectionView)
+        collectionView.fillInSuperview()
+    }
+
+    // MARK: - Public methods
+
+    public func reloadData() {
+        collectionView.reloadData()
+    }
+
+    public func calculateSize(constrainedTo width: CGFloat) -> CGSize {
+        // Is this needed?
+        CGSize(width: width, height: 76)
```

Comment:
To answer my own question: yes, it's needed😄 I'll update this one.

Reply:
haha okay, thought it wasn't 😂 ignore my other comment. 
where is it needed though?

Reply:
We need it to set the height constraint. I guess you're right that it could be a static property, though.

https://github.com/finn-no/FinniversKit/pull/1031/files/383516799c0bf7134986671352cafffec2c5c38a#diff-090d55fc20efa74c5f30120c8c3671ddf31993a116b636ee48870619665ff430R168